frontend/src/component/incomingWebhooks/IncomingWebhooksEvents/IncomingWebhooksEventsModal.tsx

```diff
@@ -0,0 +1,180 @@
+import { Button, Link, styled } from '@mui/material';
+import { SidebarModal } from 'component/common/SidebarModal/SidebarModal';
+import { IIncomingWebhook } from 'interfaces/incomingWebhook';
+import { useIncomingWebhookEvents } from 'hooks/api/getters/useIncomingWebhookEvents/useIncomingWebhookEvents';
+import { Suspense, lazy } from 'react';
+import FormTemplate from 'component/common/FormTemplate/FormTemplate';
+import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
+import { SidePanelList } from 'component/common/SidePanelList/SidePanelList';
+import { formatDateYMDHMS } from 'utils/formatDate';
+import { useLocationSettings } from 'hooks/useLocationSettings';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+
+const LazyReactJSONEditor = lazy(
+    () => import('component/common/ReactJSONEditor/ReactJSONEditor'),
+);
+
+const StyledHeader = styled('div')(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    marginBottom: theme.fontSizes.mainHeader,
+}));
+
+const StyledHeaderRow = styled('div')({
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    width: '100%',
+});
+
+const StyledHeaderSubtitle = styled('div')(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    marginTop: theme.spacing(2),
+    fontSize: theme.fontSizes.smallBody,
+}));
+
+const StyledDescription = styled('p')(({ theme }) => ({
+    color: theme.palette.text.secondary,
+}));
+
+const StyledTitle = styled('h1')({
+    fontWeight: 'normal',
+});
+
+const StyledForm = styled('form')({
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%',
+});
+
+const StyledButtonContainer = styled('div')(({ theme }) => ({
+    marginTop: 'auto',
+    display: 'flex',
+    justifyContent: 'flex-end',
+    paddingTop: theme.spacing(4),
+}));
+
+interface IIncomingWebhooksEventsModalProps {
+    incomingWebhook?: IIncomingWebhook;
+    open: boolean;
+    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+    onOpenConfiguration: () => void;
+}
+
+export const IncomingWebhooksEventsModal = ({
+    incomingWebhook,
+    open,
+    setOpen,
+    onOpenConfiguration,
+}: IIncomingWebhooksEventsModalProps) => {
+    const { uiConfig } = useUiConfig();
+    const { locationSettings } = useLocationSettings();
+    const { incomingWebhookEvents, hasMore, loadMore, loading } =
+        useIncomingWebhookEvents(incomingWebhook?.id, 20, {
+            refreshInterval: 5000,
+        });
+
+    if (!incomingWebhook) {
```

Comment:
under which circumstances would this occur?

Reply:
Cool, thanks for the explanation!

Reply:
What would the drawbacks of conditionally rendering the Modal be?

Reply:
That could also work, but if we're using `ConditionallyRender`, the component should be returned in a callback, so something like this:

```tsx
<ConditionallyRender
    condition={Boolean(selectedIncomingWebhook)}
    show={() => (
        <IncomingWebhooksModal
            incomingWebhook={selectedIncomingWebhook!}
            open={modalOpen}
            setOpen={setModalOpen}
            newToken={(token: string) => {
                setNewToken(token);
                setTokenDialog(true);
            }}
            onOpenEvents={() => {
                setModalOpen(false);
                setEventsModalOpen(true);
            }}
        />
    )}
/>
```

TypeScript also doesn't know that `show` only happens when the condition is `true`, so you also need to add the `!` in `selectedIncomingWebhook!`. But if you do this, I believe it's safe to assume that the `incomingWebhook` prop is no longer optional since it's always going to be present.

While this could be a valid option and sometimes easier, the main drawbacks for me at this stage are noisy code at the parent level and less consistency with the rest of our code base.

Effectively, what we have now, checking for `if (!incomingWebhook) return null` inside the component, is the same as a conditional render, but shifting the responsibility of handling this gracefully to the component itself and not the parent.